### PR TITLE
[Blazor] Enable custom icu loading

### LINKED ISSUE #83585: [browser][icu] Enable custom ICU loading for Blazor
Apply the same feature we have in WASM browser app (https://github.com/dotnet/runtime/issues/79987), for Blazor.

Connected issues:
https://github.com/dotnet/runtime/issues/81552

### PR DESCRIPTION
**Pull request by ilonatommy**
Fixes https://github.com/dotnet/runtime/issues/83585.
Needs to be merged together with a change in aspnetcore repo https://github.com/dotnet/aspnetcore/pull/47419 and a change in documentation in runtime repo https://github.com/dotnet/runtime/pull/83925.

It is an extension of the custom ICU feature implemented already in WASM apps, to Blazor apps. WASM change was introduced here: https://github.com/dotnet/runtime/pull/80421.

Changes from user perspective:
- they can decrease the size of runtime by loading a custom ICU file instead of pre-prepared icu files from the runtime pack. They need to prepare the file beforehand and provide info about its location to get loaded. Then, they add this information as MSBuild property `<BlazorIcuDataFileName>icudt_path.dat</BlazorIcuDataFileName>`.
- new warnings when they combine globalization/icu flags incorrectly.

Changes:
- new icu mode: custom (3),
- `resources.runtime` from `blazor.boot.json` in the custom case will contain only one, custom icu file name + hash.

cc @lewing, @maraf

### CODE REVIEW

File: src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets [hunk 1]
```diff
@@ -122,16 +122,27 @@ Copyright (c) .NET Foundation. All rights reserved.
     </ItemGroup>
   </Target>
 
-  <Target Name="_ResolveBlazorWasmConfiguration">
+  <Target Name="_ResolveGlobalizationConfiguration">
+    <Warning Condition="'$(InvariantGlobalization)' == 'true' AND '$(BlazorWebAssemblyLoadAllGlobalizationData)' == 'true'" Text="%24(BlazorWebAssemblyLoadAllGlobalizationData) has no effect when %24(InvariantGlobalization) is set to true." />
+    <Warning Condition="'$(InvariantGlobalization)' == 'true' AND '$(BlazorIcuDataFileName)' == 'true'" Text="%24(BlazorIcuDataFileName) has no effect when %24(InvariantGlobalization) is set to true." />
```
Comment (maraf) [inline src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets:127]: `BlazorIcuDataFileName` points to a file, right? Should we test if for "not empty"?

File: src/BlazorWasmSdk/Tasks/ComputeBlazorBuildAssets.cs [hunk 2]
```diff
@@ -18,6 +18,8 @@ public class ComputeBlazorBuildAssets : Task
         [Required]
         public ITaskItem[] Candidates { get; set; }
 
+        public ITaskItem CustomIcuCandidate { get; set; } = new TaskItem();
```
Comment (javiercn) [inline src/BlazorWasmSdk/Tasks/ComputeBlazorBuildAssets.cs:21]: Is there a reason to initialize this property, typically we just check against the default if not present.

**Review by javiercn — APPROVED** [review]
Looks great!

File: src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets [hunk 3]
```diff
@@ -122,16 +122,27 @@ Copyright (c) .NET Foundation. All rights reserved.
...
+    <Warning Condition="'$(BlazorWebAssemblyLoadAllGlobalizationData)' == 'true' AND '$(BlazorIcuDataFileName)' != ''" Text="%24(BlazorIcuDataFileName) has no effect when %24(BlazorWebAssemblyLoadAllGlobalizationData) is set to true." />
+    <PropertyGroup>
+      <_BlazorWebAssemblyLoadAllGlobalizationData Condition="'$(InvariantGlobalization)' != 'true'">$(BlazorWebAssemblyLoadAllGlobalizationData)</_BlazorWebAssemblyLoadAllGlobalizationData>
+      <_BlazorWebAssemblyLoadAllGlobalizationData Condition="'$(_BlazorWebAssemblyLoadAllGlobalizationData)' == ''">false</_BlazorWebAssemblyLoadAllGlobalizationData>
+      <_BlazorIcuDataFileName Condition="'$(InvariantGlobalization)' != 'true' AND '$(BlazorWebAssemblyLoadAllGlobalizationData)' != 'true'">$(BlazorIcuDataFileName)</_BlazorIcuDataFileName>
```
Comment (maraf) [inline src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets:132]: Can we validate that the file name starts with "icudt" and throw warning/error if not?

Comment (ilonatommy) [reply src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets:132]: Good idea!

File: src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets (hunk 1 above)
Comment (ilonatommy) [reply src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets:127]: Ofc you're right, correcting

**Pull request merged**